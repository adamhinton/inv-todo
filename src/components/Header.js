import styled from "styled-components";

const Header = (props) => {
  const { setTodoList } = props;

  return (
    <StyledHeader>
      <Styledh1>Inventora Tasks</Styledh1>
      <StyledDeleteAllButton
        onClick={() => {
          setTodoList([]);
        }}
      >
        Delete All
      </StyledDeleteAllButton>
    </StyledHeader>
  );
};

export default Header;

const StyledHeader = styled.header`
  width: 100%;
  display: flex;
  justify-content: flex-start;
  margin-left: 5%;
  border-bottom: 1px solid lightgrey;
  margin-bottom: 3%;
  align-items: center;
  max-width: 100%;

  @media (max-width: 450px) {
    flex-direction: column;
    text-align: center;
  }
`;

const Styledh1 = styled.h1`
  margin-right: 300px;

  @media (max-width: 750px) {
    margin-right: 25%;
  }

  @media (max-width: 450px) {
    width: 80%;
    margin: 0;
  }
`;

const StyledDeleteAllButton = styled.button`
  border: none;
  background-color: #0d98ba;
  width: 150px;
  color: white;
  height: 50%;
  font-size: 1.25rem;
  padding: 15px 30px;
  border-radius: 10px;
`;
