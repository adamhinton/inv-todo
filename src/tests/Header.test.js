import Header from "../components/Header";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

test("[1] Header renders without errors", () => {
  render(<Header />);
});

test("[2] Header and Delete All button appear on screen", () => {
  render(<Header />);

  const h1 = screen.getByText("Inventora Tasks");
  const deleteAllButton = screen.getByText("Delete All");

  expect(h1).toBeVisible();
  expect(deleteAllButton).toBeVisible();
});

test("[3] Matches snapshot from 5.29.22", () => {
  const component = renderer.create(<Header />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
