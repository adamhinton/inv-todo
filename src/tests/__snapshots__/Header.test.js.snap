// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[3] Matches snapshot from 5.29.22 1`] = `
<header
  className="sc-bczRLJ iUswKO"
>
  <h1>
    Inventora Tasks
  </h1>
  <button
    className="sc-gsnTZi cLPkNS"
    onClick={[Function]}
  >
    Delete All
  </button>
</header>
`;
